### PR TITLE
Blog - Implementing OAuth 2.0 Flow in Non-Web Clients

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks

### CODE REVIEW

**Comment by msali123** [comment]
https://6434322fb20578471dc260ce--ephemeral-narwhal-3ac3b2.netlify.app/blog/oauth-2.0-in-non-web-clients/

@jalletto

**Comment by msali123** [comment]
[Preview Link](https://6435422152ac264b02e65b1d--ephemeral-narwhal-3ac3b2.netlify.app/blog/oauth-2-in-non-web-clients/)
@balapriyac

File: blog/_posts/2023-06-05-oauth-2-in-non-web-clients.md [hunk 1]
```diff
@@ -0,0 +1,691 @@
...
+
+Recall that the browser flow does not occur on the non-web client. So, the non-web client needs to know the status of the authorization request. The status can be known by polling the authorization server in an access_token request. The polling occurs in the `get_access_token_from_login_code` function.
+
+The `get_access_token_from_login_code` function involves the following steps:
+
+- Makes an API call to the Facebook access token endpoint with the device code and access token passed as query params.
+- Calls the `_handle_error_from_login_code` function if there is an error. If the error is not a `PendingActionError` it raises an exception that terminates the function execution and if otherwise, delays the code by the 5s using `await asyncio.sleep()`.
+- The polling is achieved by using [recursion](https://users.cs.utah.edu/~germain/PPS/Topics/recursion.html). You can also use a loop to achieve the polling.
+
+<div class="notice--info">
```
Comment (balapriyac) [inline blog/_posts/2023-06-05-oauth-2-in-non-web-clients.md:610]:
```suggestion
<div class="notice--big--primary">
```

**Comment by msali123** [comment]
https://6436d4e30c386e13e9144d0e--ephemeral-narwhal-3ac3b2.netlify.app/blog/oauth-2-in-non-web-clients/

@balapriyac changes done

**Comment by balapriyac** [comment]
@jalletto this is ready to merge!

**Review by jalletto — APPROVED** [review]

**Pull request merged**